### PR TITLE
Week3-Part1

### PR DESCRIPTION
**Pull request by gulyener**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,17 +26,91 @@
...
+
+    try {
+      const contributors = await fetchJSON(repo.contributors_url);
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul);
+        const a = createAndAppend('a', li, { href: contributor.html_url, target: '_blank' });
+        const table = createAndAppend('table', a);
+        const tbody = createAndAppend('tbody', table);
+        const tr1 = createAndAppend('tr', tbody);
+        createAndAppend('img', tr1, { src: contributor.avatar_url });
```
Comment (remarcmij) [inline homework/index.js]: This created invalid HTML. A `tr` element can only have `td` and `th` children. 

```html
<ul id="list-contributions">
  <li>Contributors:</li>
  <li>
    <a href="https://github.com/gijscor" target="_blank">
      <table>
        <tbody>
          <tr>
            <img src="https://avatars0.githubusercontent.com/u/15912395?v=4" />  <-- invalid HTML
            <td>gijscor</td>
            <td>47</td>
          </tr>
        </tbody>
      </table></a
    >
  </li>
</ul>
```

File: homework-classes/App.js [hunk 2]
```diff
@@ -19,12 +19,24 @@ class App {
...
 
     try {
       const repos = await Util.fetchJSON(url);
       this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const select = Util.createAndAppend('select', header);
+      this.repos.sort((a, b) => a.name().localeCompare(b.name()));
+      this.repos.forEach((repo, index) => {
+        Util.createAndAppend('option', select, {
+          text: repo.repository.name,
```
Comment (remarcmij) [inline homework-classes/App.js]: In OOP, the data from a class (in this case the Repository class), should be considered private to that class. If another class, such as in this case the App class, it should request the Repository class to provide that data, by calling one of its methods. This principle is called 'data encapsulation' (google for it). As you can see in the Repository class, a method was provided for this purpose, to obtain the name. (You di it correctly in line 29).

So this line should read:

```js
text: repo.name(),
```

File: homework/index.js (hunk 1 above)
Comment (gulyener) [reply homework/index.js]: Thanks I fixed it

File: homework-classes/Contributor.js [hunk 3]
```diff
@@ -13,7 +13,16 @@ class Contributor {
...
-    // TODO: replace the next line with your code.
-    Util.createAndAppend('pre', container, { text: JSON.stringify(this.contributor, null, 2) });
+    const li = Util.createAndAppend('li', container);
+    const a = Util.createAndAppend('a', li, {
+      href: this.contributor.html_url,
+      target: '_blank',
+    });
+    const table = Util.createAndAppend('table', a);
+    const tbody = Util.createAndAppend('tbody', table);
+    const tr1 = Util.createAndAppend('tr', tbody);
```
Comment (remarcmij) [inline homework-classes/Contributor.js]: No real need for this `1` suffix in the name `tr1`, is there?

**Pull request closed without merging**